modules/sbo-data-model.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// *openshift-docs/applications/connecting_applications_to_services/exposing-binding-data-from-a-service
+[id="sbo-data-model_{context}"]
+= Data model
+
+Service binding annotations follow the convention:
+
+[source,yaml]
+----
+service.binding(/<NAME>)?: "<VALUE>|(path=<JSONPATH_TEMPLATE>(,objectType=<OBJECT_TYPE)?(,elementType=<ELEMENT_TYPE>)?(,sourceKey=<SOURCE_KEY>)?(,sourceValue=<SOURCE_VALUE>)?)"
+----
+where:
+[horizontal]
+`<NAME>`:: Specifies the name under which the binding value is going to be exposed. It can be excluded only when the `objectType` parameter is set to `Secret` or `ConfigMap`.
+`<VALUE>`:: Specifies the constant value exposed, when no `path` is set.
+
+This section explains the data model used in the annotations and OLM descriptors. The data model is the same for custom resource definition (CRD) or custom resource (CR) annotations and Operator Lifecycle Manager (OLM) descriptors, but the syntax is different, which is explained in the respective sections.
+
+The data model provides the details on the allowed values and semantic for the `path`, `elementType`, `objectType`, `sourceKey`, and `sourceValue` parameters.
+
+.Parameters and their descriptions
```

Comment:
just checked https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/intro-expose-binding.html#_data_model and seeing that the section differs from the content in this PR. Why is that so?

Reply:
The content in the https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/intro-expose-binding.html#_data_model is quite complex for the user interpretation. Hence we have simplified it. I will raise a JIRA issue and also a PR to do some edits in the upstream content. The difference in the content will be addressed as part of the work.